JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/slurm/job/JobSubmission.java

```diff
@@ -1063,13 +1078,8 @@ public void deleteFolderOrFile(String folderOrFileName) throws JSchException, Sf
 	 */
 	public boolean hostAvailabilityCheck(String server, int port) {
 		boolean available = true;
-		try {
-			(new Socket(server, port)).close();
-		} catch (UnknownHostException e) {
-			available = false;
-		} catch (IOException e) {
-			available = false;
-		} catch (NullPointerException e) {
+		try (final Socket dummy = new Socket(server, port)){
```

Comment:
I would suggest checking availability on something different than IO/NullPointer exceptions. It seems to be too generic and can easily  lead to false negatives.

Reply:
I was just trying to remove the code flow where an exception thrown during the creation of the Socket could lead to the Socket not being closed, by changing it to use try-with-resources. I was not trying to fix the existing logic of the function. Looking at the code in the Socket constructor, the possibility of an IOException must be handled although I can't see an obvious source of one other than the subclass UnknownHostException. Regarding the NullPointerException I agree that this is very general also again looking into the Socket constructor's code it doesn't seem very likely that one will be thrown although the person who originally wrote the hostAvailabilityCheck method might have encountered it in a way that made this appropriate, I don't know. I could also see that a possible IllegalArgumentException exception can be thrown if the port value is out of the valid range so adding that in would probably be more appropriate. I'll make some changes to the method for you to check.

Reply:
I've added a further commit that I think answers this comment, please review it.

Reply:
Please also .close() the socket.

Reply:
The whole reason I changed this method in the first place was to remove the need to explicitly call the "close" method.
Support for "try-with-resources" was added in Java 7 to reduce mistakes with regards to closing things like files and sockets. Please read this article about using "try-with-resources" statements https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html .
I would strongly recommend that some mention of using "try-with-resources" is added to the coding standards document as it should be used wherever possible.